client/my-sites/domains/components/domain-warnings/index.jsx

```diff
@@ -815,7 +815,7 @@ export class DomainWarnings extends PureComponent {
 			( domain ) => hasPendingGSuiteUsers( domain ) || isPendingGSuiteTOSAcceptance( domain )
 		);
 
-		if ( domains.length === 0 ) {
+		if ( domains.length === 0 || ! this.props.isCompact ) {
```

Comment:
This isn't the best place to limit rendering on the domain mgmt list. This is rather central.

It seems to me that it's better to limit rendering at the places where the component is called, like [here for instance](https://github.com/Automattic/wp-calypso/blob/fc1c2034600e49a8dbe0172d9894adac6b766a4b/client/my-sites/domains/domain-management/list/index.jsx#L93)